mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -674,12 +674,13 @@ object LogisticRegressionModel extends MLReadable[LogisticRegressionModel] {
 
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.format("parquet").load(dataPath)
-        .select("numClasses", "numFeatures", "intercept", "coefficients").head()
+        .select("numClasses", "numFeatures", "intercept", "coefficients")
+      val row = MLUtils.convertVectorColumnsToML(data, "coefficients").head()
```

Comment:
I think we should call convert vector columns first and then call `select` to ensure the ordering of columns. There is no contact on keeping the column ordering between mappings.